File: serveur/src/main/java/well_tennis_club/projet/disponibility/DisponibilityService.java
package well_tennis_club.projet.disponibility;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.UUID;

@Service
public class DisponibilityService {
    private final DisponibilityRepository disponibilityRepository;
    @Autowired
    public DisponibilityService(DisponibilityRepository repository){this.disponibilityRepository = repository;}

    public List<DisponibilityEntity> getAllDisponibilities(){return disponibilityRepository.findAll();}

    public DisponibilityEntity createDisponibility(DisponibilityEntity entity){return disponibilityRepository.save(entity);}

    public DisponibilityEntity updateDisponibility(DisponibilityEntity entity){return disponibilityRepository.save(entity);}

    public void deleteDisponibility(DisponibilityEntity entity){disponibilityRepository.delete(entity);}

    public DisponibilityEntity getDisponibilityById(UUID id){return disponibilityRepository.findById(id).orElse(null);}
}

File: well_tennis_affichage_vu2/src/services/DisponibilityService.js
// src/services/DisponibilityService.js
import apiService from './apiService';

export default {
	getAllDisponibility() {
		return apiService.getData('/disponibility');
	},

	getDisponibilityById(id) {
		return apiService.getData(`/disponibility/${id}`);
	},

	 createDisponibility(disponibilityData) {
		return apiService.post("/disponibility", disponibilityData);
	},

	updateDisponibility(id, DisponibilityData) {
		return apiService.patch(`/disponibility/${id}`, DisponibilityData);
	},

	deleteDisponibility(id) {
		return apiService.delete(`/disponibility/${id}`);
	},
};

